### PR TITLE
Avoid some crashes

### PR DESCRIPTION
**Pull request by dan-alcantara-neeva**
* Add boundary check when the user is scrolling through their archived tabs.  Because the list is being paged in asynchronously, you can get into a situation where the list is wiped while the UI is trying to index into the list, resulting in an `OutOfBoundsException`.

* Catch and drop `IllegalArgumentException`s thrown by WebLayer when trying to initialize Cookie Cutter.  Play doesn't provide us the error message, so until we switch to Firebase or some other crash logging framework we can't really debug this.  Because Cookie Cutter is optional, it should be "fine" to do nothing when it can't be turned on for a particular tab.

### Pointers to help with review
* Figma link (if this is a UI change)
* Link to where the change is discussed (e.g. meeting notes, design docs)

### How was this tested?
- [ ] I tested this manually
- [X] I added automated tests
- [ ] Existing automated tests cover some of my changes
- [ ] Existing automated tests cover all of my changes

### Manual test cases
Because these crashes are so random, we can't really test these.

## Issues addressed
* Fixes https://github.com/neevaco/neeva-android/issues/1002
* Ignores https://github.com/neevaco/neeva-android/issues/1003

### CODE REVIEW

**Comment by Kokobe** [comment]
Add the firebase logging as an issue TO-DO?

File: app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt [hunk 1]
```diff
@@ -111,6 +111,33 @@ class ProcessedArchivedTabsTest : BaseTest() {
...
+            // We should get a valid item.
+            expectThat(getTabData(4)).isEqualTo(tabDataList[2])
+
+            // Say that the PagedList has been emptied out.
+            every { lazyTabs.itemCount } returns 0
+
+            // Confirm that we going out of bounds doesn't cause a crash and returns null.
+            expectThat(getTabData(4)).isNull()
+        }
+    }
```
Comment (Kokobe) [inline app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt:139]: Add a test for your new boundary condition where tabs.itemCount == 0 in getTabData()

**Comment by dan-alcantara-neeva** [comment]
> Add the firebase logging as an issue TO-DO?

There's a difference between TODOs and issues:
* TODOs are _supposed_ to be handled quickly as follow up.
* Issues are filed when something will take longer, or will be punted for a bit.

Logging has been on the backburner for a bit, but Chung has started looking into adding support for another framework that might tangentially give us better crash logs.

**Comment by dan-alcantara-neeva** [comment]
> Add the firebase logging as an issue TO-DO?

There's a difference between TODOs and issues:
* TODOs are _supposed_ to be handled quickly as follow up.
* Issues are filed when something will take longer, or will be punted for a bit.

Logging has been on the backburner for a bit, but Chung has started looking into adding support for another framework that might tangentially give us better crash logs.

File: app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt (hunk 1 above)
Comment (dan-alcantara-neeva) [reply app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt:139]: Doesn't really work here; the 0th tab item would be a header for the items below it, so there's nothing to index into.

Comment (Kokobe) [reply app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt:148]: I'm confused, why can't you do: 
```
 val tabDataList = createTabDataList(listOf())
```

Comment (dan-alcantara-neeva) [reply app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt:148]: Didn't really want to, but duplicated a bit of the test above it and null checked everything.  Ideally unit tests wouldn't overlap so much, but you can't avoid it sometimes if you want to be clear about what's happening.

Comment (dan-alcantara-neeva) [reply app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt:148]: > I'm confused, why can't you do:
> 
> ```
>  val tabDataList = createTabDataList(listOf())
> ```

That's not what the test is checking.  The `ProcessedTabData` processes the `TabList` once and then uses that to index into the `PagedList` of `Tab`s.  The app crashes because the indices into the `PagedList` get out of sync and we still try to access the item at a particular index in the `PagedList` after the list has been cleared out.

Comment (Kokobe) [reply app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt:148]: Oh I get it now!

**Review by Kokobe — APPROVED** [review]

File: app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt (hunk 1 above)
Comment (dan-alcantara-neeva) [reply app/src/test/java/com/neeva/app/cardgrid/archived/ProcessedArchivedTabsTest.kt:148]: Added some comments to the `ProcessedArchivedTabs` @Kokobe ; can you take a look and see if it helps?

**Review by Kokobe — APPROVED** [review]
new comment documentation looks good! Thank you for adding it
